### PR TITLE
[luajit] Support UNIX

### PR DESCRIPTION
**Pull request by JackBoosY**
Needs to install autoconfig and autobuild in UNIX.

Related: #5782.

Note: this port does not contain any features.

### CODE REVIEW

**Comment by JackBoosY** [comment]
I found that Linux / OSX was not tested. What's going wrong?

**Comment by JackBoosY** [comment]
@ras0219-msft Still not tested.

File: ports/luajit/portfile.cmake [hunk 1]
```diff
@@ -1,5 +1,3 @@
-include(vcpkg_common_functions)
-
 if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
     message(FATAL_ERROR "LuaJIT currently only supports being built for desktop")
 endif()
```
Comment (PhoebeHui) [inline ports/luajit/portfile.cmake]: Please replace it with vcpkg_fail_port_install(ON_TARGET "uwp")

**Review by PhoebeHui — APPROVED** [review]

**Comment by vicroms** [comment]
I've removed **luajit** from the CI exclusion list, as there are no documented conflicts.

**Comment by JackBoosY** [comment]
OSX build failure:
```
Undefined symbols for architecture x86_64:
  "__Unwind_DeleteException", referenced from:
      _lj_err_unwind_dwarf in libluajit.a(lj_err.o)
  "__Unwind_GetCFA", referenced from:
      _lj_err_unwind_dwarf in libluajit.a(lj_err.o)
  "__Unwind_RaiseException", referenced from:
      _lj_err_throw in libluajit.a(lj_err.o)
  "__Unwind_SetGR", referenced from:
      _lj_err_unwind_dwarf in libluajit.a(lj_err.o)
  "__Unwind_SetIP", referenced from:
      _lj_err_unwind_dwarf in libluajit.a(lj_err.o)
ld: symbol(s) not found for architecture x86_64
```

From [official workaround](https://github.com/LuaJIT/LuaJIT/issues/449),  I think the PR should wait for #9569 to merge.

**Comment by JackBoosY** [comment]
Temporary close this PR, will continue this work in the future.

**Pull request closed without merging**